pydantic_ai_slim/pydantic_ai/agent/__init__.py

```diff
@@ -370,6 +370,14 @@ def __init__(
             _utils.Option[Sequence[Tool[AgentDepsT] | ToolFuncEither[AgentDepsT, ...]]]
         ] = ContextVar('_override_tools', default=None)
 
+        # Prompt overrides (experimental)
+        self._override_instructions: ContextVar[_utils.Option[str | None]] = ContextVar(
```

Comment:
Could we support lists and functions as well, like on `Agent.__init__`?

https://github.com/pydantic/pydantic-ai/blob/a2883d26c538a4033143ce10198b37fcebd5669d/pydantic_ai_slim/pydantic_ai/agent/__init__.py#L166-L169